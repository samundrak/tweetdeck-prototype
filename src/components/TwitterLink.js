import React from 'react';
import PropTypes from 'prop-types';

const TwitterLink = ({ handle, children }) => (
  <a href={`https://twitter.com/${handle}`} target="_blank">
    {children || `@${handle}`}
  </a>
);
TwitterLink.defaultProps = {
  children: <div />,
};
TwitterLink.propTypes = {
  handle: PropTypes.string.isRequired,
  children: PropTypes.node,
};
export default TwitterLink;
