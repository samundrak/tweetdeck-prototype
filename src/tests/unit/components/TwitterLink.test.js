import React from 'react';
import { shallow } from 'enzyme';
import TwitterLink from '../../../components/TwitterLink';

describe('Test TwitterLink', () => {
  it('should render TwitterLink', () => {
    const handle = 'kardnumas';
    const wrapper = shallow(<TwitterLink handle={handle}>{handle}</TwitterLink>);
    expect(wrapper.text()).toBe(handle);
  });
  it('should test snapshot', () => {
    const handle = 'kardnumas';
    const wrapper = shallow(<TwitterLink handle={handle}>{handle}</TwitterLink>);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
