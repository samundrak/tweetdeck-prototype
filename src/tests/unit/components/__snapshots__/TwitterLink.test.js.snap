// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TwitterLink should test snapshot 1`] = `"<a href=\\"https://twitter.com/kardnumas\\" target=\\"_blank\\">kardnumas</a>"`;
